1-Draft/RFCnnnn-Alias-Policy.md

```diff
@@ -0,0 +1,108 @@
+---
+RFC: nnnn
+Author: Ilya Sazonov
+Status: Draft
+Area:
+Comments Due: 6/30/2018
+---
+
+# The document purpose
+
+The purpose of this document is to establish a new policy for the creation and use of aliases to ensure:
+
+* Reliable, predictable behavior of scripts and modules
+* Increase readability and maintainability of scripts and modules
+* UX enhancements in interactive sessions
+
+# Problem description
+
+1. Windows PowerShell has many aliases that conflict with the names of standard Unix utilities.
+
+    These aliases was removed from PowerShell Core.
+    This causes a backward compatibility issue when Windows users do not find the familiar aliases in the interactive session.
+    It also disrupts the operation of custom scripts that use these aliases.
+
+1. Now there are dozens of modules developed by MSFT and thousands of modules of third developers. This causes a name conflict.
+
+    Developers try to create unique cmdlet names.
+    They can use module prefixes to resolve name conflicts.
+
+    On the other hand, developers of the best intentions can complement their modules cmdlet aliases.
+    At the same time, users can create their own aliases.
+
+    With a large number of existing modules and cmdlets and a wide variety of custom aliases, these short names inevitably conflict with each other.
+
+    Moreover, the behavior becomes unpredictable because each subsequent download module or script can change any alias.
+    Either the user can destroy a script behavior by creating a new alias or by deleting an existing one.
+
+    It is even more unpredictable and dangerous to create scripts in the assumption that there are some predefined (standard) aliases because they can be changed by the user or script at any time.
```

Comment:
Statements like this keep getting repeated -- but they are not relevant.

The fear of aliases being changed is **irrelevant** except when **PowerShell changes it's defaults**.

In fact, aliases are _no more likely to be changed_ than the commands that they point to --- it is even possible to register a function with a name that clobbers the so-called "fully qualified" command name.  There are, in fact, several modules in the gallery which replace or "clobber" built-in commands (including commands which have built-in aliases) -- but I don't know of any which alter any of the built-in aliases.

Reply:
To which I'm dealing with right now, PowerCLI and PSCX conflict with built-in cmdlets and ones I wrote. There is no easy-clean way to stop that because of Autoloading of Modules. I see no difference with Aliases in this context. 

The reliability of the environment from importing new modules or scripts is a *separate issue* and *RFC*.

Reply:
> To which I'm dealing with right now, PowerCLI and PSCX conflict with built-in cmdlets and ones I wrote. There is no easy-clean way to stop that because of Autoloading of Modules. I see no difference with Aliases in this context.
> 
> The reliability of the environment from importing new modules or scripts is a **separate issue** and **RFC**.

This is pretty much the issue that is currently making me blow a gasket.
Hyper-V, Power-CLI and VirtualMachineManager all use some overlapping aliases, which either cause imports to fail, or commands to not run as expected. Hence why I am having to remove all aliases.